### PR TITLE
Improve Utils.parseURL to handle relative URLs

### PR DESCRIPTION
**Pull request by brianjmiller**
This fixes more statement fetching, particularly as used by the prototypes identified by https://github.com/RusticiSoftware/TinCan_Prototypes/issues/36.

### CODE REVIEW

**Comment by brianjmiller** [comment]
@brianrogers can you review? Was going to have @garemoko do it but he is on vacation for the next few days. (Andrew happy to have your review upon return if this hasn't cleared.)

**Comment by garemoko** [comment]
I'm back on Monday so happy to review then. One quick comment @brianjmiller now though - could we have a test where the protocol is localhost? I had a problem the other day launching the prototypes on my local machine from Moodle on my local machine and assumed it was my Moodle plugin doing something wrong but actually I'm wondering if that regex doesn't like localhost.

**Comment by brianjmiller** [comment]
@garemoko I may have noticed that recently with the prototypes too, but I'm pretty sure I determined it was unrelated (we should still look into it). Either way I added the tests here since they are reasonable, particularly for single word domains and they pass.

**Comment by garemoko** [comment]
Thanks. Seeing that test jogged me to realize what i was doing wrong in the Moodle plugin settings too. I was trying to launch ```localhost``` instead of ```http://localhost``` so not an issue in TinCanJS at all.

**Comment by garemoko** [comment]
Reviewing this now, so assigning to me.

File: src/Utils.js [hunk 1]
```diff
@@ -236,15 +236,23 @@ TinCan client library
         /**
         @method parseURL
         @param {String} url
+        @param {Object} [options]
+            @param {Boolean} [options.allowRelative] Option to allow relative URLs
         @return {Object} Object of values
         @private
         */
-        parseURL: function (url) {
+        parseURL: function (url, options) {
```
Comment (garemoko) [inline src/Utils.js]: Other functions use 'cfg' for what seems to me to be a similar purpose to 'options' here. Is there a good reason not to use cfg here for consistency?

**Comment by garemoko** [comment]
Aside from the cfg vs. options comment above, this looks good to me. I did a build and tested with the prototypes to confirm this fixes https://github.com/RusticiSoftware/TinCan_Prototypes/issues/36 (and doesn't appear to break anything else).

**Comment by brianjmiller** [comment]
I hate to say it but the code is terribly inconsistent around the 'cfg' / 'options' thing with it looking like the majority of code uses `cfg` as the variable but the API docs use `options` so I set it to be like that.

**Pull request merged**